Update bootstrap script from upstream gnulib to automatically fetch from repository if needed

See also: https://lists.gnu.org/archive/html/bug-gnulib/2020-08/msg00150.html

--- ./bootstrap-funclib.sh.orig	2023-11-20 18:29:45.640064746 +0100
+++ ./bootstrap-funclib.sh	2023-11-20 18:35:57.451241555 +0100
@@ -534,6 +534,11 @@
         fi
 
         trap - HUP INT PIPE TERM
+
+      elif test -n "$GNULIB_REVISION" \
+           && ! git --git-dir="$gnulib_path"/.git cat-file \
+                commit "$GNULIB_REVISION"; then
+        git --git-dir="$gnulib_path"/.git fetch
       fi
     fi
     GNULIB_SRCDIR=$gnulib_path
@@ -546,6 +551,10 @@
   # XXX Should this be done if $use_git is false?
   if test -d "$GNULIB_SRCDIR"/.git && test -n "$GNULIB_REVISION" \
      && ! git_modules_config submodule.gnulib.url >/dev/null; then
+    if ! git --git-dir="$GNULIB_SRCDIR"/.git cat-file \
+         commit "$GNULIB_REVISION"; then
+      git --git-dir="$GNULIB_SRCDIR"/.git fetch
+    fi
     (cd "$GNULIB_SRCDIR" && git checkout "$GNULIB_REVISION") || cleanup_gnulib
   fi
 
